workers/ohsome_quality_analyst/ohsome/client.py

```diff
@@ -20,45 +20,38 @@ async def query(
     ratio: bool = False,
 ) -> Optional[Dict]:
     """Query ohsome API endpoint with filter."""
-    if endpoint is not None:
-        url = OHSOME_API + endpoint
-    else:
-        url = OHSOME_API + layer.endpoint
-
-    if ratio:
-        url = url + "/ratio"
-        data = {"bpolys": bpolys, "filter": layer.filter, "filter2": layer.ratio_filter}
-    else:
-        data = {"bpolys": bpolys, "filter": layer.filter}
-
-    if time is not None:
-        data["time"] = time
-
+    url = build_url(layer, endpoint, ratio)
+    data = build_data_dict(layer, bpolys, time, ratio)
     logging.info("Query ohsome API.")
-    logging.debug("Query data: " + json.dumps(data, indent=4, sort_keys=True))
+    logging.debug("Query URL: " + url)
+    logging.debug("Query filter: " + layer.filter)
+    logging.debug("Query data: " + json.dumps(data))
+    return await query_ohsome_api(url, data)
+
 
-    # set custom timeout as ohsome API takes a long time to send an answer
-    timeout = httpx.Timeout(5, read=600)
+async def query_ohsome_api(url: str, data: dict):
```

Comment:
I broke it down to make it easier to test.

Also when ohsomehex endpoints will be used we probably need to implement some kind of logic to decide which API will be queried. This could be done in the 'query' function.